Lib/fontPens/flattenPen.py

```diff
@@ -42,26 +42,44 @@ def _lineTo(self, pt):
             self.currentPt = pt
             return
         step = 1.0 / maxSteps
-        factors = range(0, maxSteps + 1)
-        for i in factors[1:]:
-            self.otherPen.lineTo(interpolatePoint(self.currentPt, pt, i * step))
+        for factor in range(1, maxSteps + 1):
+            self.otherPen.lineTo(interpolatePoint(self.currentPt, pt, factor * step))
         self.currentPt = pt
 
     def _curveToOne(self, pt1, pt2, pt3):
+        falseCurve = (pt1 == self.currentPt) and (pt2 == pt3)
+        if falseCurve:
+            self._lineTo(pt3)
+            return
         est = estimateCubicCurveLength(self.currentPt, pt1, pt2, pt3) / self.approximateSegmentLength
         maxSteps = int(round(est))
-        falseCurve = (pt1 == self.currentPt) and (pt2 == pt3)
-        if maxSteps < 1 or falseCurve:
+        if maxSteps < 1:
             self.otherPen.lineTo(pt3)
             self.currentPt = pt3
             return
         step = 1.0 / maxSteps
-        factors = range(0, maxSteps + 1)
-        for i in factors[1:]:
-            pt = getCubicPoint(i * step, self.currentPt, pt1, pt2, pt3)
+        for factor in range(1, maxSteps + 1):
+            pt = getCubicPoint(factor * step, self.currentPt, pt1, pt2, pt3)
             self.otherPen.lineTo(pt)
         self.currentPt = pt3
 
+    def _qCurveToOne(self, pt1, pt2):
+        falseCurve = (pt1 == self.currentPt) or (pt1 == pt2)
+        if falseCurve:
+            self._lineTo(pt2)
+            return
+        est = estimateQuadraticCurveLength(self.currentPt, pt1, pt2) / self.approximateSegmentLength
```

Comment:
@jenskutilek wait.. I merged too fast. Why are you still using `estimateQuadraticCurveLength` if you now have `calcQuadraticArcLength` from fonttools?

Reply:
Yeah, I noticed just now that the new function is not used ...